pyproject.toml

```diff
@@ -3,81 +3,115 @@ requires = ["pdm-backend"]
 build-backend = "pdm.backend"
 
 [tool.pdm.build]
-includes = [
-    "stac_model",
-]
+includes = ["stac_model"]
 
 [project]
 authors = [
-    {name = "Ryan Avery", email = "ryan@wherobots.com"},
-    {name = "Francis Charette-Migneault", email = "francis.charette-migneault@crim.ca"},
+  { name = "Ryan Avery", email = "ryan@wherobots.com" },
+  { name = "Francis Charette-Migneault", email = "francis.charette-migneault@crim.ca" },
 ]
-license = {text = "Apache Software License 2.0"}
-requires-python = "<4.0,>=3.10"
+license = { text = "Apache Software License 2.0" }
+requires-python = "<4.0,>=3.11"
 dependencies = [
-    "typer<1.0.0,>=0.9.0",
-    "rich>=13.7.0,<15.0.0",
-    "pydantic<3.0.0,>=2.6.3",
-    "pydantic-core<3,>=2",
-    "pystac<2.0.0,>=1.9.0",
-    "shapely<3,>=2",
-    "jsonschema<5.0.0,>=4.21.1",
-    "pip>=25.0.0",
+  "typer<1.0.0,>=0.9.0",
+  "rich>=13.7.0,<15.0.0",
+  "pydantic<3.0.0,>=2.6.3",
+  "pydantic-core<3,>=2",
+  "pystac<2.0.0,>=1.9.0",
+  "shapely<3,>=2",
+  "jsonschema<5.0.0,>=4.21.1",
+  "pip>=25.0.0",
+]
+optional-dependencies.torch = ["torch>=2.8.0.dev,<2.9.0", "torchvision>=0.21"]
+optional-dependencies.torch-cu126 = [
+  "pytorch-triton; sys_platform=='linux'",
+  "torch>=2.8.0.dev,<2.9.0; sys_platform=='linux'",
+  "torchgeo; sys_platform=='linux'",
+  "torchvision>=0.21,<1; sys_platform=='linux'",
 ]
```

Comment:
I think the workaround to still allow Python 3.10 is to indicate explicit `python_version >= 3.11` conditions on the `torchgeo` references (see https://python-poetry.org/docs/dependency-specification/#multiple-constraints-project). 

Obviously, a user that wants to work with `torchgeo` would have no choice to respect that Python 3.11 constraint, but packages not using this optional dependency it should theoretically still be allowed Python 3.10.

Reply:
Addressed this here: https://github.com/wherobots/mlm/pull/3